lib/services/livesync/android-platform-livesync-service.ts

```diff
@@ -0,0 +1,55 @@
+import {PlatformLiveSyncServiceBase} from "./platform-livesync-service-base";
+
+class AndroidPlatformLiveSyncService extends PlatformLiveSyncServiceBase {
+	constructor(_liveSyncData: ILiveSyncData,
+		protected $devicesService: Mobile.IDevicesService,
+		protected $mobileHelper: Mobile.IMobileHelper,
+		protected $logger: ILogger,
+		protected $options: ICommonOptions,
+		protected $deviceAppDataFactory: Mobile.IDeviceAppDataFactory,
+		protected $fs: IFileSystem,
+		protected $injector: IInjector,
+		protected $projectFilesManager: IProjectFilesManager,
+		protected $projectFilesProvider: IProjectFilesProvider,
+		protected $liveSyncProvider: ILiveSyncProvider) {
+		super(_liveSyncData, $devicesService, $mobileHelper, $logger, $options, $deviceAppDataFactory, $fs, $injector, $projectFilesManager, $projectFilesProvider, $liveSyncProvider);
+	}
+
+	public fullSync(): IFuture<void> {
+		return (() => {
+			let appIdentifier = this.liveSyncData.appIdentifier;
+			let platform = this.liveSyncData.platform;
+			let projectFilesPath = this.liveSyncData.projectFilesPath;
+			let canExecute = this.getCanExecuteAction(platform, appIdentifier);
+			let action = (device: Mobile.IDevice): IFuture<void> => {
+				return (() => {
+					let platformLiveSyncService = this.resolvePlatformLiveSyncService(platform, device);
```

Comment:
I completely agree with you, but I decided to leave this change to another PR because this one is becoming quite large. If you are OK I will leave it as is and change it later in some other PR